app/views/handlers/preview_questions_pdf.py

```diff
@@ -1,10 +1,26 @@
 import io
 
+from app.data_models import QuestionnaireStore
+from app.questionnaire import QuestionnaireSchema
 from app.views.handlers.pdf_response import PDFResponse
 from app.views.handlers.view_preview_questions import ViewPreviewQuestions
 
 
+class PreviewNotEnabledException(Exception):
+    pass
+
+
 class PreviewQuestionsPDF(PDFResponse, ViewPreviewQuestions):
+    def __init__(
+        self,
+        schema: QuestionnaireSchema,
+        questionnaire_store: QuestionnaireStore,
+        language: str,
+    ):
+        super().__init__(schema, questionnaire_store, language)
+        if not schema.json.get("preview_questions"):
+            raise PreviewNotEnabledException(404)
```

Comment:
You don't need it in PDF/Preview since they both end up calling the same class. Raise the exception at the lowest level, then anything that uses it will get the check.

Reply:
I am not sure here, `PreviewContext` and `PreviewQuestionsPDF` don't seem to be descendants of the same classes? In these two I raise this exception now.

Reply:
They both call out to `PreviewContext` right?

Reply:
Ok, calling it at the lowest level now, in `PreviewContext`.